fauna/client/client.py

```diff
@@ -574,6 +638,82 @@ def close(self):
       self._stream.close()
 
 
+@dataclass
+class ChangeFeedPage:
+  events: List[Any]
+  cursor: str
+  stats: QueryStats
+
+  def __iter__(self) -> Iterator[Any]:
+    return iter(self.events)
+
+
+class ChangeFeedIterator:
+  """A class to provide an iterator on top of change feed pages."""
+
+  def __init__(self, http: HTTPClient, headers: Dict[str, str],
+               endpoint: str, max_attempts: int, max_backoff: int,
+               opts: ChangeFeedOptions, token: StreamToken):
+    self._http = http
+    self._headers = headers
+    self._endpoint = endpoint
+    self._max_attempts = opts.max_attempts or max_attempts
+    self._max_backoff = opts.max_backoff or max_backoff
+    self._request = {"token": token.token}
+    self._is_done = False
+
+    if opts.page_size is not None:
+      self._request["page_size"] = opts.page_size
+
+    if opts.cursor is not None:
+      self._request["cursor"] = opts.cursor
+    elif opts.start_ts is not None:
+      self._request["start_ts"] = opts.start_ts
```

Comment:
This means that the cursor will take precedence if both a start time and cursor are provided. Can we consider raising an error if both are provided instead?

We don't do that for streams, which I think we should be consistent with. The cursor options have not been published, so we have a chance to change the behavior for stream config too.

Reply:
Okay sounds good.